api/src/main/java/org/apache/iceberg/metrics/DefaultMetricsContext.java

```diff
@@ -42,4 +42,21 @@ public <T extends Number> Counter<T> counter(String name, Class<T> type, Unit un
   public Timer timer(String name, TimeUnit unit) {
     return new DefaultTimer(unit);
   }
+
+  @Override
+  public <T extends Number> Gauge<T> gauge(String name, Class<T> type) {
+    if (Integer.class.equals(type)) {
+      return (Gauge<T>) new IntGauge();
+    }
+
+    if (Long.class.equals(type)) {
+      return (Gauge<T>) new LongGauge();
+    }
+    throw new IllegalArgumentException(String.format("Counter for type %s is not supported", type.getName()));
```

Comment:
nit: should this say `Gauge` instead of `Counter`?

Reply:
good catch. fixed.